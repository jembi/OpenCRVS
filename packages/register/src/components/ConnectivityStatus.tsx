import * as React from 'react'
import { Online, Offline } from '@opencrvs/components/lib/icons'
import styled from '@register/styledComponents'

const StyledOnline = styled.div`
  position: absolute;
  top: 32px;
  right: 245px;
`

const StyledOffline = styled.div`
  position: absolute;
  top: 32px;
  right: 245px;
`

const ConnectivityStatus = () => {
  return navigator.onLine ? (
    <StyledOnline>
      <Online />
    </StyledOnline>
  ) : (
    <StyledOffline>
      <Offline />
    </StyledOffline>
  )
}

export default ConnectivityStatus
