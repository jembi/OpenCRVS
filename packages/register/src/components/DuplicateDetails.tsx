import * as React from 'react'
import { Box, Chip } from '@opencrvs/components/lib/interface'
import styled from 'styled-components'
import {
  StatusGray,
  StatusOrange,
  StatusGreen,
  StatusRejected,
  StatusCollected,
  Delete
} from '@opencrvs/components/lib/icons'
import { PrimaryButton } from '@opencrvs/components/lib/buttons'
import { defineMessages, injectIntl, InjectedIntlProps } from 'react-intl'
import { connect } from 'react-redux'
import { goToTab as goToTabAction } from 'navigation'
import { REVIEW_EVENT_PARENT_FORM_TAB } from 'navigation/routes'
import Moment from 'react-moment'
import { getRejectionReasonDisplayValue } from 'views/SearchResult/SearchResult'

export enum Event {
  BIRTH = 'BIRTH',
  DEATH = 'DEATH'
}

export enum Action {
  DECLARED = 'DECLARED',
  REJECTED = 'REJECTED',
  REGISTERED = 'REGISTERED',
  CERTIFIED = 'CERTIFIED'
}

interface IProps {
  id: string
  duplicateContextId: string
  data: {
    id: string
    dateOfApplication: string
    trackingId: string
    event: Event
    child: {
      name: string
      dob: string
      gender: string
    }
    mother?: {
      name: string
      dob: string
      id: string
    }
    father?: {
      name: string
      dob: string
      id: string
    }
    regStatusHistory: Array<{
      action: Action
      date: string
      usersName: string
      usersRole: string
      office: string
      reasons?: string
    }>
  }
  notDuplicateHandler?: () => void
  rejectHandler?: () => void
  gotoTab: typeof goToTabAction
}

const messages = defineMessages({
  name: {
    id: 'register.duplicates.details.name',
    defaultMessage: 'Name',
    description: 'Name label'
  },
  dob: {
    id: 'register.duplicates.details.dob',
    defaultMessage: 'D.o.B.',
    description: 'Date of birth label'
  },
  gender: {
    id: 'register.duplicates.details.gender',
    defaultMessage: 'Gender',
    description: 'Gender label'
  },
  dateOfApplication: {
    id: 'register.duplicates.details.dateOfApplication',
    defaultMessage: 'Date of application',
    description: 'Date of application label'
  },
  trackingId: {
    id: 'register.duplicates.details.trackingId',
    defaultMessage: 'Tracking ID',
    description: 'Tracking ID label'
  },
  notDuplicate: {
    id: 'register.duplicates.details.notDuplicate',
    defaultMessage: 'Not a duplicate?',
    description: 'A Question which is a link: Not a duplicate?'
  },
  mother: {
    id: 'register.duplicates.details.mother',
    defaultMessage: 'Mother',
    description: 'Mother section label'
  },
  father: {
    id: 'register.duplicates.details.father',
    defaultMessage: 'Father',
    description: 'Father section label'
  },
  id: {
    id: 'register.duplicates.details.id',
    defaultMessage: 'ID',
    description: 'ID Label'
  },
  applicationState: {
    id: 'register.duplicates.details.applicationState',
    defaultMessage: 'Application {action} on',
    description: 'A label to describe when the application was actioned on'
  },
  by: {
    id: 'register.duplicates.details.by',
    defaultMessage: 'By',
    description: 'Label for By (the person who performed the action)'
  },
  reason: {
    id: 'register.duplicates.details.reason',
    defaultMessage: 'Reason',
    description: 'Label for Reason the application was rejected'
  },
  review: {
    id: 'register.duplicates.details.review',
    defaultMessage: 'Review',
    description: 'A label from the review button'
  },
  BIRTH: {
    id: 'register.duplicates.details.birthEvent',
    defaultMessage: 'Birth',
    description: 'A label from the birth event'
  },
  DEATH: {
    id: 'register.duplicates.details.deathEvent',
    defaultMessage: 'Death',
    description: 'A label from the death event'
  },
  application: {
    id: 'register.duplicates.details.application',
    defaultMessage: 'application',
    description: 'A label for application'
  },
  DECLARED: {
    id: 'register.duplicates.details.submitted',
    defaultMessage: 'submitted',
    description: 'A label for submitted'
  },
  REJECTED: {
    id: 'register.duplicates.details.rejected',
    defaultMessage: 'rejected',
    description: 'A label for rejected'
  },
  REGISTERED: {
    id: 'register.duplicates.details.registered',
    defaultMessage: 'registered',
    description: 'A label for registered'
  },
  CERTIFIED: {
    id: 'register.duplicates.details.certified',
    defaultMessage: 'certified',
    description: 'A label for certified'
  },
  rejectLinkText: {
    id: 'register.duplicates.details.reject',
    defaultMessage: 'Reject',
    description: 'A label for reject link'
  }
})

const DetailsBox = styled(Box).attrs<{ id: string; currentStatus: string }>({})`
  border-top: ${({ theme }) => ` 4px solid ${theme.colors.expandedIndicator}`};
  ${({ currentStatus }) =>
    currentStatus === 'rejected' ? `box-shadow: none` : ''}
`

const Separator = styled.div`
  border-top: 1px solid ${({ theme }) => theme.colors.background};
  margin: 24px -25px 24px -25px;
`

const DetailTextContainer = styled.div`
  display: flex;
  justify-content: space-between;
`

const DetailText = styled.div`
  flex: 1;
  font-family: ${({ theme }) => theme.fonts.regularFont};
  color: ${({ theme }) => theme.colors.copy};
  font-size: 16px;
`

const DetailTextSplitContainer = styled(DetailText)`
  display: flex;
  justify-content: stretch;
  cursor: pointer;
`

const Link = styled.a`
  font-family: ${({ theme }) => theme.fonts.regularFont};
  color: ${({ theme }) => theme.colors.primary};
  font-size: 16px;
  text-decoration: underline;
`

const TagContainer = styled.div`
  display: flex;
`

const ListContainer = styled.div`
  display: flex;
  flex-direction: column;
`

const ListItem = styled.div`
  display: flex;
  margin-bottom: 16px;
`

const ListStatusContainer = styled.span`
  margin: 4px 8px;
`

const RejectApplication = styled.a`
  font-family: ${({ theme }) => theme.fonts.regularFont};
  color: ${({ theme }) => theme.colors.danger};
  text-decoration: underline;
  cursor: pointer;
  margin-left: 60px;
  svg {
    margin-right: 15px;
  }
  @media (max-width: ${({ theme }) => theme.grid.breakpoints.sm}px) {
    margin-left: 30px;
  }
`
const ConditionalSeparator = styled(Separator)`
  display: none;
  @media (max-width: ${({ theme }) => theme.grid.breakpoints.sm}px) {
    border: 0px;
    display: block;
  }
`

class DuplicateDetailsClass extends React.Component<
  IProps & InjectedIntlProps
> {
  normalizeAction(action: string) {
    if (action === 'DECLARED') {
      return 'application'
    }

    return action
  }

  renderStatusIcon(action: string) {
    switch (action) {
      case 'DECLARED':
        return <StatusOrange />
      case 'REGISTERED':
        return <StatusGreen />
      case 'REJECTED':
        return <StatusRejected />
      case 'CERTIFIED':
        return <StatusCollected />
      default:
        return <StatusGray />
    }
  }

  render() {
    const currentStatus = this.props.data.regStatusHistory.slice(-1)[0].action
    const {
      data,
      intl,
      notDuplicateHandler,
      rejectHandler,
      duplicateContextId
    } = this.props

    return (
      <DetailsBox id={`detail_box_${data.id}`} currentStatus={currentStatus}>
        <DetailTextContainer>
          <DetailText>
            <b>{intl.formatMessage(messages.name)}:</b> {data.child.name}
            <br />
            <b>{intl.formatMessage(messages.dob)}:</b> {data.child.dob}
            <br />
            <b>{intl.formatMessage(messages.gender)}:</b> {data.child.gender}
            <br />
            <b>{intl.formatMessage(messages.dateOfApplication)}:</b>{' '}
            <Moment format="DD-MM-YYYY">{data.dateOfApplication}</Moment>
            <br />
            <b>{intl.formatMessage(messages.trackingId)}:</b> {data.trackingId}
            <br />
            <br />
          </DetailText>
          {notDuplicateHandler && (
            <Link
              id={`not_duplicate_link_${data.id}`}
              onClick={notDuplicateHandler}
            >
              {intl.formatMessage(messages.notDuplicate)}
            </Link>
          )}
        </DetailTextContainer>
        <DetailTextSplitContainer>
          {data.mother && (
            <DetailText>
              <b>{intl.formatMessage(messages.mother)}</b>
              <br />
              <b>{intl.formatMessage(messages.name)}:</b> {data.mother.name}
              <br />
              <b>{intl.formatMessage(messages.dob)}:</b> {data.mother.dob}
              <br />
              <b>{intl.formatMessage(messages.id)}:</b> {data.mother.id}
              <br />
            </DetailText>
          )}
          {data.father && (
            <DetailText>
              <b>{intl.formatMessage(messages.father)}</b>
              <br />
              <b>{intl.formatMessage(messages.name)}:</b> {data.father.name}
              <br />
              <b>{intl.formatMessage(messages.dob)}:</b> {data.father.dob}
              <br />
              <b>{intl.formatMessage(messages.id)}:</b> {data.father.id}
              <br />
            </DetailText>
          )}
        </DetailTextSplitContainer>
        <Separator />
        <TagContainer>
          <Chip
            status={<StatusGray />}
            text={intl.formatHTMLMessage(messages[data.event.toUpperCase()])}
          />
          <Chip
            status={this.renderStatusIcon(currentStatus)}
            text={this.normalizeAction(currentStatus)}
          />
        </TagContainer>
        <Separator />
        <ListContainer>
          {data.regStatusHistory.map((status, i) => (
            <ListItem key={i}>
              <ListStatusContainer>
                {this.renderStatusIcon(status.action)}
              </ListStatusContainer>
              <DetailText>
                <b>
                  {intl.formatMessage(messages.applicationState, {
                    action: intl.formatMessage(messages[status.action])
                  })}
                  :
                </b>{' '}
                {status.date}
                <br />
                <b>{intl.formatMessage(messages.by)}:</b> {status.usersName}
                <br />
                {status.usersRole}
                <br />
                {status.office}
                <br />
                {status.action === Action.REJECTED && status.reasons && (
                  <>
                    <b>{intl.formatMessage(messages.reason)}:</b>{' '}
                    {status.reasons
                      .split(',')
                      .map(reason =>
                        intl.formatMessage(
                          getRejectionReasonDisplayValue(reason)
                        )
                      )
                      .join(', ')}
                  </>
                )}
                <br />
              </DetailText>
            </ListItem>
          ))}
        </ListContainer>
        {currentStatus === Action.DECLARED && (
          <>
            <Separator />
            <PrimaryButton
              id={`review_link_${data.id}`}
              onClick={() => {
                this.props.gotoTab(
                  REVIEW_EVENT_PARENT_FORM_TAB,
                  data.id,
                  'review',
                  'birth',
                  '',
                  { duplicate: true, duplicateContextId }
                )
              }}
            >
              {intl.formatMessage(messages.review)}
            </PrimaryButton>
            {rejectHandler && (
              <>
                <ConditionalSeparator />
                <RejectApplication
                  id={`reject_link_${data.id}`}
                  onClick={rejectHandler}
                >
                  <Delete />
                  {intl.formatMessage(messages.rejectLinkText)}
                </RejectApplication>
              </>
            )}
          </>
        )}
      </DetailsBox>
    )
  }
}

export const DuplicateDetails = connect(
  null,
  {
    gotoTab: goToTabAction
  }
)(injectIntl<IProps>(DuplicateDetailsClass))
