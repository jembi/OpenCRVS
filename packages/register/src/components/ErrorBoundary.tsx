import * as React from 'react'
import * as Sentry from '@sentry/browser'
import styled from '@register/styledComponents'

const ErrorMessage = styled.h1`
  text-align: center;
`

interface IErrorInfo extends React.ErrorInfo {
  [key: string]: string
}

export class ErrorBoundary extends React.Component {
  state = { error: null }

  componentDidCatch(error: Error, errorInfo: IErrorInfo) {
    this.setState({ error })
    Sentry.withScope(scope => {
      Object.keys(errorInfo).forEach(key => {
        scope.setExtra(key, errorInfo[key])
      })
      Sentry.captureException(error)
    })
  }

  render() {
    if (this.state.error) {
      if (
        window.location.hostname !== 'localhost' &&
        window.location.hostname !== '127.0.0.1'
      ) {
        Sentry.showReportDialog()
      }

      // We could render fallback UI here
      return <ErrorMessage>Something went wrong...</ErrorMessage>
    } else {
      // when there's not an error, render children untouched
      return this.props.children
    }
  }
}
