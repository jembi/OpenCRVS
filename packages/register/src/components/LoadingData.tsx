import * as React from 'react'
import {
  injectIntl,
  InjectedIntlProps,
  defineMessages,
  FormattedMessage
} from 'react-intl'
import styled from 'styled-components'
import { Modal } from '@opencrvs/components/lib/interface'
import { IStoreState } from 'store'
import { connect } from 'react-redux'
import { Spinner } from '@opencrvs/components/lib/interface'
import { getOfflineDataLoaded } from 'offline/selectors'
import { PrimaryButton } from '@opencrvs/components/lib/buttons'

const StyledModal = styled(Modal)`
  z-index: 4;
`

const StyledSpinner = styled(Spinner)`
  margin: 50px auto;
`
export const messages = defineMessages({
  loadingOfflineData: {
    id: 'app.loading.data.loadingOfflineData',
    defaultMessage:
      'We are loading some essential data, so that you can submit applications offline.  Please wait ...',
    description: 'Loading offline data modal'
  },
  offlineDataLoaded: {
    id: 'app.loading.data.offlineDataLoaded',
    defaultMessage: 'OpenCRVS is ready for use.',
    description: 'Loading successful data modal'
  },
  continueButton: {
    id: 'app.loading.data.continueButton',
    defaultMessage: 'Continue',
    description: 'Continue button text'
  }
})

interface IProps {
  show: boolean
  offlineDataLoaded: boolean
  handleClose: () => void
}

const getLoadingText = (
  loaded: boolean
): FormattedMessage.MessageDescriptor => {
  return loaded ? messages.offlineDataLoaded : messages.loadingOfflineData
}

const LoadingDataComponent = ({
  show,
  handleClose,
  offlineDataLoaded,
  intl
}: IProps & InjectedIntlProps) => {
  return (
    <StyledModal
      title={intl.formatMessage(getLoadingText(offlineDataLoaded))}
      actions={[
        <PrimaryButton
          key="offline_data"
          id="offline_data_continue"
          disabled={!offlineDataLoaded}
          onClick={handleClose}
        >
          {intl.formatMessage(messages.continueButton)}
        </PrimaryButton>
      ]}
      show={show}
      handleClose={handleClose}
    >
      {!offlineDataLoaded && <StyledSpinner id="loadingOfflineDataSpinner" />}
    </StyledModal>
  )
}

export const LoadingData = connect(
  (state: IStoreState, ownProps) => ({
    offlineDataLoaded: getOfflineDataLoaded(state)
  }),
  null
)(injectIntl<IProps>(LoadingDataComponent))
