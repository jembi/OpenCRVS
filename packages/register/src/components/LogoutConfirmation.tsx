import * as React from 'react'
import { injectIntl, InjectedIntlProps, defineMessages } from 'react-intl'
import styled from 'styled-components'
import { Modal } from '@opencrvs/components/lib/interface'
import { PrimaryButton } from '@opencrvs/components/lib/buttons'

const ConfirmButton = styled.a`
  text-decoration: underline;
  color: ${({ theme }) => theme.colors.primary};
  cursor: pointer;
`
const StyledModal = styled(Modal)`
  z-index: 4;
`
export const messages = defineMessages({
  noButton: {
    id: 'app.logout.modal.no',
    defaultMessage: 'No',
    description: 'No button text on logout modal'
  },
  yesButton: {
    id: 'app.logout.modal.yes',
    defaultMessage: 'Yes',
    description: 'Yes button on logout modal'
  },
  editApplicationConfirmationTxt: {
    id: 'app.logout.modal.confirmationText',
    defaultMessage: 'Are you sure you wish to log out?',
    description: 'Logout modal confirmation text'
  }
})

interface IProps {
  show: boolean
  handleYes: () => void
  handleClose: () => void
}

const LogoutConfirmationComponent = ({
  show,
  handleYes,
  handleClose,
  intl
}: IProps & InjectedIntlProps) => {
  return (
    <StyledModal
      title={intl.formatMessage(messages.editApplicationConfirmationTxt)}
      actions={[
        <PrimaryButton key="confirm" id="logout_confirm" onClick={handleYes}>
          {intl.formatMessage(messages.yesButton)}
        </PrimaryButton>,
        <ConfirmButton key="close" id="logout_close" onClick={handleClose}>
          {intl.formatMessage(messages.noButton)}
        </ConfirmButton>
      ]}
      show={show}
      handleClose={handleClose}
    >
      <></>
    </StyledModal>
  )
}

export const LogoutConfirmation = injectIntl(LogoutConfirmationComponent)
