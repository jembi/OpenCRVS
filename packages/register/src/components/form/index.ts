export * from '@register/components/form/FormFieldGenerator'
export * from '@register/components/form/FormTabs'
export * from '@register/components/form/ViewHeaderWithTabs'
export * from '@register/components/form/FormList'
