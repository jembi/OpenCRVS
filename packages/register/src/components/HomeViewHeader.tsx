import * as React from 'react'
import {
  IViewHeadingProps,
  ViewHeading
} from '@register/components/ViewHeading'
import { Header } from '@opencrvs/components/lib/interface'
import { ActionList } from '@opencrvs/components/lib/buttons'
import styled from '@register/styledComponents'
import { TopMenu } from '@register/components/TopMenu'
import { Logo } from '@opencrvs/components/lib/icons'
import ConnectivityStatus from '@register/components/ConnectivityStatus'
import { BodyContent } from '@opencrvs/components/lib/layout'

const HeaderWrapper = styled(Header)`
  display: block;
  justify-content: flex-end;
  padding-bottom: 50px;
  min-height: 280px;

  /* stylelint-disable */
  + ${ActionList} {
    /* stylelint-enable */
    margin-top: -80px;
  }
`

const StyledLogo = styled(Logo)`
  margin: -71px auto auto 20px;
`

export class HomeViewHeader extends React.Component<IViewHeadingProps> {
  render() {
    const { title, description, id } = this.props
    return (
      <HeaderWrapper>
        <BodyContent>
          <ConnectivityStatus />
          <TopMenu hideBackButton={true} />
          <StyledLogo />
          <ViewHeading {...{ title, description, id }} />
          {this.props.children}
        </BodyContent>
      </HeaderWrapper>
    )
  }
}
