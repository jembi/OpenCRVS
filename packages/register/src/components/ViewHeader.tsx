import * as React from 'react'

import { Header } from '@opencrvs/components/lib/interface'
import { ActionList } from '@opencrvs/components/lib/buttons'
import styled from '@register/styledComponents'
import { TopMenu } from '@register/components/TopMenu'
import { Logo } from '@opencrvs/components/lib/icons'
import {
  ViewHeading,
  IViewHeadingProps
} from '@register/components/ViewHeading'
import ConnectivityStatus from '@register/components/ConnectivityStatus'
import { BodyContent } from '@opencrvs/components/lib/layout'

const HeaderWarapper = styled(Header)`
  display: block;
  justify-content: flex-end;
  padding-bottom: 50px;

  /* stylelint-disable */
  + ${ActionList} {
    /* stylelint-enable */
    margin-top: -50px;
  }
`

export class ViewHeader extends React.Component<IViewHeadingProps> {
  render() {
    const {
      title,
      description,
      breadcrumb,
      hideBackButton,
      id,
      ...otherProps
    } = this.props

    return (
      <HeaderWarapper {...otherProps}>
        <BodyContent>
          <ConnectivityStatus />
          {hideBackButton && <Logo />}
          <TopMenu hideBackButton={hideBackButton} />
          <ViewHeading {...{ title, description, breadcrumb, id }} />
          {this.props.children}
        </BodyContent>
      </HeaderWarapper>
    )
  }
}
