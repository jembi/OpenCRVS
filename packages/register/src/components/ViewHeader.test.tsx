import * as React from 'react'
import { createShallowRenderedComponent } from '@register/tests/util'
import { ViewHeader } from '@register/components/ViewHeader'

describe('view header component', () => {
  const testComponent = createShallowRenderedComponent(
    <ViewHeader
      id="test_heaader"
      title="Test title"
      description="Test description"
    />
  )

  it('renders without crashing', () => {
    expect(testComponent).toMatchSnapshot()
  })
})

describe('view header component with logo', () => {
  const hideBackButton = true
  const testComponent = createShallowRenderedComponent(
    <ViewHeader
      id="test_heaader"
      title="Test title"
      description="Test description"
      hideBackButton={hideBackButton}
    />
  )

  it('renders without crashing', () => {
    expect(testComponent).toMatchSnapshot()
  })
})
