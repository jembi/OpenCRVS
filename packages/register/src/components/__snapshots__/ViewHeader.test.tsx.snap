// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view header component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViewHeader
    description="Test description"
    id="test_heaader"
    title="Test title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div>
        <ConnectivityStatus />
        <Connect(InjectIntl(TopMenuComponent))
          hideBackButton={undefined}
        />
        <ViewHeading
          breadcrumb={undefined}
          description="Test description"
          id="test_heaader"
          title="Test title"
        />
      </styled.div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ConnectivityStatus />,
          undefined,
          <Connect(InjectIntl(TopMenuComponent))
            hideBackButton={undefined}
          />,
          <ViewHeading
            breadcrumb={undefined}
            description="Test description"
            id="test_heaader"
            title="Test title"
          />,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "hideBackButton": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "breadcrumb": undefined,
            "description": "Test description",
            "id": "test_heaader",
            "title": "Test title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div>
          <ConnectivityStatus />
          <Connect(InjectIntl(TopMenuComponent))
            hideBackButton={undefined}
          />
          <ViewHeading
            breadcrumb={undefined}
            description="Test description"
            id="test_heaader"
            title="Test title"
          />
        </styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <ConnectivityStatus />,
            undefined,
            <Connect(InjectIntl(TopMenuComponent))
              hideBackButton={undefined}
            />,
            <ViewHeading
              breadcrumb={undefined}
              description="Test description"
              id="test_heaader"
              title="Test title"
            />,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "hideBackButton": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "breadcrumb": undefined,
              "description": "Test description",
              "id": "test_heaader",
              "title": "Test title",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          undefined,
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`view header component with logo renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViewHeader
    description="Test description"
    hideBackButton={true}
    id="test_heaader"
    title="Test title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div>
        <ConnectivityStatus />
        <Unknown />
        <Connect(InjectIntl(TopMenuComponent))
          hideBackButton={true}
        />
        <ViewHeading
          breadcrumb={undefined}
          description="Test description"
          id="test_heaader"
          title="Test title"
        />
      </styled.div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ConnectivityStatus />,
          <Unknown />,
          <Connect(InjectIntl(TopMenuComponent))
            hideBackButton={true}
          />,
          <ViewHeading
            breadcrumb={undefined}
            description="Test description"
            id="test_heaader"
            title="Test title"
          />,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "hideBackButton": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "breadcrumb": undefined,
            "description": "Test description",
            "id": "test_heaader",
            "title": "Test title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div>
          <ConnectivityStatus />
          <Unknown />
          <Connect(InjectIntl(TopMenuComponent))
            hideBackButton={true}
          />
          <ViewHeading
            breadcrumb={undefined}
            description="Test description"
            id="test_heaader"
            title="Test title"
          />
        </styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <ConnectivityStatus />,
            <Unknown />,
            <Connect(InjectIntl(TopMenuComponent))
              hideBackButton={true}
            />,
            <ViewHeading
              breadcrumb={undefined}
              description="Test description"
              id="test_heaader"
              title="Test title"
            />,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "hideBackButton": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "breadcrumb": undefined,
              "description": "Test description",
              "id": "test_heaader",
              "title": "Test title",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          undefined,
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
